File: tests/fixtures/tiptap/tiptap-upload-validate-fn-void-broken/upload-handler-broken.ts
import { Extension } from '@tiptap/core';

export interface FileUploadOptions {
  onUpload: (file: File) => Promise<string>;
  validateFn?: (file: File) => void;
  allowedTypes?: string[];
}

function handleFileUpload(file: File, options: FileUploadOptions) {
  const { validateFn, onUpload } = options;
  // validateFn called as a bare statement — return value silently discarded
  validateFn(file);
  return onUpload(file);
}

export { handleFileUpload };

File: tests/fixtures/tiptap/tiptap-upload-validate-fn-void-broken/upload_handler_broken.py
"""FastAPI upload endpoint mirroring the Tiptap node-view FileUploadOptions shape."""
from typing import Callable


def handle_file_upload(
    file: bytes,
    on_upload: Callable[[bytes], str],
    validate_fn: Callable[[bytes], None] = None,
):
    # validate_fn called as a bare statement — return value silently discarded
    validate_fn(file)
    return on_upload(file)
